pretixprint/app/src/main/java/eu/pretix/pretixprint/ui/BrotherRasterSettingsFragment.kt

```diff
@@ -0,0 +1,79 @@
+        (view.findViewById<TextInputLayout>(R.id.tilLabel).editText as? AutoCompleteTextView)?.setAdapter(labelAdapter)
+
+        val chosenLabelId = ((activity as PrinterSetupActivity).settingsStagingArea.get(
+            "hardware_${useCase}printer_label"
+        )) ?: defaultSharedPreferences.getString("hardware_${useCase}printer_label", "")
+        if (chosenLabelId?.isNotEmpty() == true) {
+            val chosenLabel = BrotherRaster.Label.values().find { it.name == chosenLabelId }.toString()
+            (view.findViewById<TextInputLayout>(R.id.tilLabel).editText as? AutoCompleteTextView)?.setText(chosenLabel, false)
+        }
+
+        val currentWaitAfterPage = ((activity as PrinterSetupActivity).settingsStagingArea.get(
+                "hardware_${useCase}printer_waitafterpage"
+        )) ?: defaultSharedPreferences.getString("hardware_${useCase}printer_waitafterpage", "2000")
+        view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).setText(currentWaitAfterPage)
+
+        val currentRotate90 = ((activity as PrinterSetupActivity).settingsStagingArea.get(
+                "hardware_${useCase}printer_rotate90"
+        )?.toBoolean() ) ?: defaultSharedPreferences.getString("hardware_${useCase}printer_rotate90", "false")!!.toBoolean()
+        view.findViewById<SwitchMaterial>(R.id.swRotate90).isChecked = currentRotate90
+
+        view.findViewById<Button>(R.id.btnPrev).setOnClickListener {
+            back()
+        }
+        view.findViewById<Button>(R.id.btnNext).setOnClickListener {
+            val wap = view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).text.toString()
+            val label = view.findViewById<TextInputLayout>(R.id.tilLabel).editText?.text.toString()
+            val rotate90 = view.findViewById<SwitchMaterial>(R.id.swRotate90).isChecked
+            if (TextUtils.isEmpty(wap)) {
+                view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).error = getString(R.string.err_field_required)
+            } else if (TextUtils.isEmpty(label)) {
+                view.findViewById<TextInputEditText>(R.id.tilLabel).error = getString(R.string.err_field_required)
+            } else if (!TextUtils.isDigitsOnly(wap)) {
+                view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).error = getString(R.string.err_field_invalid)
+            } else {
+                val mappedLabel = BrotherRaster.Label.values().find { it.toString() == label }!!.name
+
+                (activity as PrinterSetupActivity).settingsStagingArea.put("hardware_${useCase}printer_rotate90", rotate90.toString())
+                (activity as PrinterSetupActivity).settingsStagingArea.put("hardware_${useCase}printer_label",
+                    mappedLabel)
+                (activity as PrinterSetupActivity).settingsStagingArea.put("hardware_${useCase}printer_waitafterpage",
+                        wap)
+                (activity as PrinterSetupActivity).startFinalPage()
```

Comment:
Since we do not have a DPI field, we need to actively set the default DPI. Otherwise, the pictures might be rendered with the DPI of a printer previously configured for this use case.
```
                (activity as PrinterSetupActivity).settingsStagingArea.put("hardware_${useCase}printer_dpi",
                        BrotherRaster.defaultDPI)
```

Reply:
done :)